### ISSUE TITLE
Allow au-compose to use binding context scope like v1

### ISSUE THREAD

**Issue by Vheissu** [aurelia/aurelia#952]
# 🙋 Feature Request

The new `<au-compose>` element is missing a few features that many (including myself) used in Aurelia v1 and dynamic composition using `<compose>`. One of those features is the ability for the compose element to inherit the parent binding context.

## 🤔 Expected Behavior

I should be able to dynamically compose a HTML file akin to Aurelia 1 `<compose view="./my-component.html"></compose>` without the need to provide a view-model or binding context. That HTML file then assumes the parent view-model as its context, allowing it to access methods and properties.

## 😯 Current Behavior

A new locked context appears to be created (like it does for all custom elements). As such, the current behaviour is that whatever you're dynamically composing is segregated from the rest of the page and does not have the binding context.

## 💁 Possible Solution

Traverse the parent scope and get the surrounding context if no view-model/context is supplied.

## 🔦 Context

Trying to dynamically compose UI like I can in Aurelia 1, specifically, the ability to dynamically compose HTML.

## 💻 Examples

In Aurelia 1 you can do this, not providing a `view-model` will pass in the current scope to the HTML view.
```
<template repeat.for="widget of widgets">
    <compose view.bind="widget.view"></compose>
</template>
```

**Comment by Vheissu** [comment]
@bigopon I just realised that @EisenbergEffect created an issue which is pretty much the same thing as this one. I'll keep both for context, but they should both help. https://github.com/aurelia/aurelia/issues/498

**Comment by bigopon** [comment]
Resolved by #1184
